core/domain/stats_services.py

```diff
@@ -40,6 +40,31 @@ def get_top_unresolved_answers_for_default_rule(exploration_id, state_name):
     }
 
 
+def get_exp_wise_unresolved_answers_count_for_default_rule(exp_ids):
+    exploration_objects = exp_services.get_multiple_explorations_by_id(exp_ids)
+    explorations_ids_list = [
+        exp_domain_objects[1].id
+        for exp_domain_objects in exploration_objects.iteritems()
+        for state_tuple in exp_domain_objects[1].states.iteritems()
+    ]
+    exps_answers_mapping = {}
+
+    for ind, statewise_answers in enumerate(
+            get_top_state_rule_answers_multi(
```

Comment:
I think it will be clearer to break this list construction out, e.g. do
`exploration_state_list = ...`
`for ind, statewise_answers in enumerate(exploration_state_list):`